components/inputs/input-color.js

```diff
@@ -0,0 +1,411 @@
+		}
+		const oldVal = this._associatedValue;
+		this._associatedValue = (typeof val === 'string') ? val.toUpperCase() : undefined;
+		this.requestUpdate('associatedValue', oldVal);
+	}
+
+	get value() {
+		if (this._value === undefined && this.disallowNone) {
+			return this.type === 'background' ? DEFAULT_VALUE_BG : DEFAULT_VALUE;
+		}
+		return this._value;
+	}
+	set value(val) {
+		if (val !== undefined && !HEX_REGEX.test(val)) {
+			throw new TypeError(`<d2l-input-color>: invalid HEX value "${val}"`);
+		}
+		const oldVal = this._value;
+		this._value = (typeof val === 'string') ? val.toUpperCase() : undefined;
+		this.requestUpdate('value', oldVal);
+	}
+
+	static get focusElementSelector() {
+		return '#opener';
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		this._validateLabel();
+	}
+
+	render() {
+
+		const label = !this.labelHidden ? html`<div class="d2l-input-label">${this._getLabel()}</div>` : nothing;
+		const tooltip = !this._opened ? html`<d2l-tooltip for="opener" for-type="label">${this._getTooltipLabel()}</d2l-tooltip>` : nothing;
+		const opener = this._getOpener();
+
+		return html`${label}${opener}${tooltip}`;
+
+	}
+
+	async updated(changedProperties) {
+
+		super.updated(changedProperties);
+
+		if (changedProperties.has('label') || changedProperties.has('type')) this._validateLabel();
+		if (changedProperties.has('launchType')) this._validateLaunchType();
+
+	}
+
+	_getLabel() {
+		if (this.label !== undefined) return this.label;
+		if (this.type === 'background') return this.localize('components.input-color.backgroundColor');
+		if (this.type === 'foreground') return this.localize('components.input-color.foregroundColor');
+		return '';
+	}
+
+	_getOpener() {
+		if (this.readonly) {
+			return html`<div class="readonly-wrapper" id="opener" role="img" tabindex="0">${this._getSwatch()}</div>`;
+		}
+		const buttonClass = {
+			'd2l-button': true,
+			'd2l-dropdown-opener': this.launchType === 'dropdown',
+			'opened': this._opened
+		};
+		const ariaLabel = this._opened ? this._getTooltipLabel() : undefined;
+		const button = html`<button id="opener" class="${classMap(buttonClass)}" aria-disabled="${ifDefined(this.disabled ? 'true' : undefined)}" aria-label="${ifDefined(ariaLabel)}" @click="${this._handleOpenDialog}">${this._getSwatch()}<d2l-icon icon="tier1:chevron-down-small"></d2l-icon></button>`;
+		if (this.launchType === 'dialog') {
+			return button;
+		}
+		return html`
+			<d2l-dropdown @d2l-dropdown-close="${this._handleClose}" @d2l-dropdown-open="${this._handleOpenDropdown}">
+				${button}
+				<d2l-dropdown-content max-width="350" mobile-tray="left" no-padding ?opened="${this._opened}">
+					<slot @d2l-input-color-dropdown-close="${this._handleCloseRequest}"><div style="padding: 10px">Currently, color inputs only support a "launch-type" of "dropdown" from within the MVC framework. Use "launch-type" of "dialog" instead.</div></slot>
```

Comment:
This is just a helpful message to folks if they try to use this in `dropdown` mode outside of a MVC page.

<img width="360" alt="Screen Shot 2023-03-01 at 10 19 40 AM" src="https://user-images.githubusercontent.com/5491151/222238152-fc3b7f92-599d-4f94-8673-a93d9f520f88.png">